confirmation_wizard/wizard/confirmation_message_wizard.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+
+
+class ConfirmationMessageWizard(models.TransientModel):
+    _name = "confirmation.message.wizard"
+    _description = "Confirmation Wizard"
+
+    text = fields.Char(string="Confirm Message", required=True)
+
+    res_ids = fields.Char()
+    res_model = fields.Char()
+
+    callback_method = fields.Char()
+    callback_params = fields.Json()
+
+    return_type = fields.Selection(
+        [
+            ("window_close", "Return Window Close Action"),
+            ("method", "Return Method"),
+        ],
+        default="window_close",
+    )
+
+    @api.model
+    def _prepare_action(self, title=None):
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "confirmation_wizard.confirmation_message_wizard_action"
+        )
+        if title:
+            action["name"] = title
+        action["context"] = self._context
+        return action
+
+    @api.model
+    def confirm_message(
+        self, text, records, title=None, method=None, callback_params=None
+    ):
+        callback_params = callback_params or {}
+        res_ids = ",".join(map(str, records.ids))
+        wizard = self.create(
+            {
+                "text": text,
+                "res_ids": res_ids,
+                "return_type": "method",
+                "res_model": records._name,
+                "callback_method": method,
+                "callback_params": callback_params,
+            }
+        )
+        return wizard._prepare_action(title)
+
+    @api.model
+    def confirm_no_action_message(self, text, title=None):
+        wizard = self.create(
+            {
+                "text": text,
+                "return_type": "window_close",
+            }
+        )
+        return wizard._prepare_action(title)
+
+    def _confirm_window_close(self):
+        return {"type": "ir.actions.act_window_close"}
+
+    def _confirm_method(self):
+        if self.res_ids:
+            res_ids = map(int, self.res_ids.split(","))
+        else:
+            res_ids = []
+        records = self.env[self.res_model].browse(res_ids)
+        if not records.exists():
+            raise models.UserError(
+                _(f"Records (IDS: {self.res_ids}) not found in model {self.res_model}.")
```

Comment:
please, don't use f-string, translations may not work: 

_("Records (IDS: '%(ids)s') not found in model '%(model)s'.")